testnet/stacks-node/src/event_dispatcher.rs

```diff
@@ -365,13 +357,15 @@ impl EventDispatcher {
         let mut events: Vec<(bool, Txid, &StacksTransactionEvent)> = vec![];
         let mut i: usize = 0;
 
-        let boot_receipts = if chain_tip.metadata.block_height == 1 {
-            Some(&self.boot_receipts)
-        } else {
-            None
-        };
+        let mut boot_receipts = vec![];
+        if chain_tip.metadata.block_height == 1 {
+            let mut boot_receipts_result = self.boot_receipts.borrow_mut();
+            if let Some(mut val) = boot_receipts_result.take() {
+                boot_receipts.append(&mut val);
```

Comment:
Append will be an expensive operation for a large genesis block -- because it will be iterating over every entry in the vec.

I think you should be able to do something like this:

```suggestion
 let mut boot_receipts = 
        if chain_tip.metadata.block_height == 1 {
            let mut boot_receipts_result = self.boot_receipts.borrow_mut();
            if let Some(val) = boot_receipts_result.take() {
                val
            } else {
                vec![]
            }
        } else {
           vec![]
        };
```